service/src/main/java/bio/terra/workspace/service/resource/ResourceValidationUtils.java

```diff
@@ -212,6 +212,21 @@ public static void validateAzureIPorSubnetName(String name) {
     }
   }
 
+  /**
+   * See
+   * https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules
+   * for azure resource rules
+   */
+  public static void validateAzureNamespace(String name) {
+    Pattern pattern = Pattern.compile("^[a-zA-Z0-9][a-zA-Z0-9-]{0,78}[a-zA-Z0-9]$");
```

Comment:
Small nit, it might be good to save the compiled pattern rather than instantiating a new one on each invocation of this method.

Reply:
updated!